### PR TITLE
Fix multiples connections to Cere,Ethereum,Polygon

### PR DESCRIPTION
**Pull request by DmitriyMolch**
https://cerenetwork.atlassian.net/browse/CBI-2046

### CODE REVIEW

**Review by AndreiNavoichyk — APPROVED** [review]
Please, confirm:
1. You've tested all flows
2. You've tested networks switching

File: src/Contexts/Adaptors/SubstrateDestinationAdaptor.tsx [hunk 1]
```diff
@@ -121,19 +118,15 @@ export const SubstrateDestinationAdaptorProvider = ({
           }
         });
       });
-      setListenerActive(unsubscribe);
+      setListenerActive(true);
     } else if (listenerActive && !depositNonce) {
-      const unsubscribeCall = async () => {
-        setListenerActive(undefined);
-      };
-      unsubscribeCall();
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/SubstrateDestinationAdaptor.tsx:129]: Why we don't need it more?

File: src/Contexts/Adaptors/EVMAdaptors/helpers.ts [hunk 2]
```diff
@@ -19,9 +19,9 @@ const getRpcProviderFromHttpUrl = (url: string) => {
       user: urlInstance.username,
       password: urlInstance.password,
     };
-    return new ethers.providers.JsonRpcProvider(urlInfo);
+    return new ethers.providers.StaticJsonRpcProvider(urlInfo);
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/EVMAdaptors/helpers.ts:22]: What is StaticJsonRpcProvider?

File: src/Contexts/Adaptors/SubstrateDestinationAdaptor.tsx (hunk 1 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/SubstrateDestinationAdaptor.tsx:129]: As discussed `listenerActive` is not called in the code.

File: src/Contexts/Adaptors/EVMAdaptors/helpers.ts (hunk 2 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/EVMAdaptors/helpers.ts:22]: `StaticJsonRpcProvider` doesn't make a call to get `chainId` per each request. So if we don't change the network then it should be fine. Here is a link for that [issue](https://github.com/ethers-io/ethers.js/issues/901)

Comment (DmitriyMolch) [reply src/Contexts/Adaptors/EVMAdaptors/helpers.ts:22]: Here is also a [documentation](https://docs.ethers.io/v5/api/providers/jsonrpc-provider/#StaticJsonRpcProvider) for this top.

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**